src/errors/__tests__/network_error.test.ts

```diff
@@ -0,0 +1,99 @@
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", "http://www.example.com");
+    const requestError = new RequestError(xhr, "foo", "bar");
+    const networkError = new NetworkError("foo", requestError);
+    expect(networkError).toBeInstanceOf(Error);
+    expect(networkError.name).toBe("NetworkError");
+    expect(networkError.type).toBe("NETWORK_ERROR");
+    expect(networkError.xhr).toBe(requestError.xhr);
+    expect(networkError.status).toBe(requestError.status);
+    expect(networkError.errorType).toBe(requestError.type);
+    expect(networkError.reason).toBe(requestError);
+    expect(networkError.code).toBe("");
+    expect(networkError.fatal).toBe(false);
+    expect(networkError.message).toBe("NetworkError () bar");
+  });
+
+  it("should be able to set it as fatal", () => {
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", "http://www.example.com");
+    const requestError = new RequestError(xhr, "foo", "bar");
+    const networkError = new NetworkError("foo", requestError, true);
+    expect(networkError).toBeInstanceOf(Error);
+    expect(networkError.name).toBe("NetworkError");
+    expect(networkError.type).toBe("NETWORK_ERROR");
+    expect(networkError.xhr).toBe(requestError.xhr);
+    expect(networkError.status).toBe(requestError.status);
+    expect(networkError.errorType).toBe(requestError.type);
+    expect(networkError.reason).toBe(requestError);
+    expect(networkError.code).toBe("");
+    expect(networkError.fatal).toBe(true);
+    expect(networkError.message).toBe("NetworkError () bar");
+  });
+
+  it("should filter in a valid error code", () => {
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", "http://www.example.com");
+    const requestError = new RequestError(xhr, "foo", "bar");
+    const networkError = new NetworkError("MEDIA_ERR_NETWORK", requestError, true);
+    expect(networkError).toBeInstanceOf(Error);
+    expect(networkError.name).toBe("NetworkError");
+    expect(networkError.type).toBe("NETWORK_ERROR");
+    expect(networkError.xhr).toBe(requestError.xhr);
+    expect(networkError.status).toBe(requestError.status);
+    expect(networkError.errorType).toBe(requestError.type);
+    expect(networkError.reason).toBe(requestError);
+    expect(networkError.code).toBe("MEDIA_ERR_NETWORK");
+    expect(networkError.fatal).toBe(true);
+    expect(networkError.message).toBe("NetworkError (MEDIA_ERR_NETWORK) bar");
+  });
+
+  it("should return false in isHttpError if not an HTTP error", () => {
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", "http://www.example.com");
+    const requestError = new RequestError(xhr, "foo", "bar");
+    const networkError = new NetworkError("MEDIA_ERR_NETWORK", requestError, true);
+    expect(networkError.isHttpError(0)).toBe(false);
+  });
+
+  /* tslint:disable max-line-length */
+  it("should return false in isHttpError if it is an HTTP error with a different code", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 72. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=439&issues=AWgpECiT_MPaBDn0IVTt&open=AWgpECiT_MPaBDn0IVTt)

[ISSUE_KEY]: AWgpECiT_MPaBDn0IVTt